skynet.go

```diff
@@ -16,21 +16,67 @@ import (
 )
 
 type (
+	// Skykey contains information about a skykey.
+	Skykey struct {
+		Skykey string `json:"skykey"`
+		Name   string `json:"name"`
+		ID     string `json:"id"`
+		Type   string `json:"type"`
+	}
+
+	// UploadData contains data to upload, indexed by filenames.
+	UploadData map[string]io.Reader
+
+	// Options structs.
+
+	// ConnectionOptions contains options used for connecting to Skynet.
+	ConnectionOptions struct {
+		// PortalURL is the URL of the portal to use.
+		PortalURL string
+		// CustomUserAgent is the custom user agent to use.
+		CustomUserAgent string
+	}
+
+	// AddSkykeyOptions contains the options used for addskykey.
+	AddSkykeyOptions struct {
+		ConnectionOptions
+		// PortalAddSkykeyPath is the relative URL path of the addskykey
+		// endpoint.
+		PortalAddSkykeyPath string
+	}
+	// CreateSkykeyOptions contains the options used for createskykey.
+	CreateSkykeyOptions struct {
+		ConnectionOptions
+		// PortalCreateSkykeyPath is the relative URL path of the createskykey
+		// endpoint.
+		PortalCreateSkykeyPath string
```

Comment:
Looking at this MR more I'm wondering if it makes more sense to rename this to `APIRoute` and add it to the `ConnectionOptions`. 

That would allow us to generalize the code even more. Unless there is a plan to move the parameters like `name`, `id`, `skykeytype` into their corresponding options struct it seems unnecessary to be defining an Options struct for each endpoint. 

Then this line becomes even more clear
```go
url := makeURL(opts.PortalURL, opts.APIRoute)
```

To be able to define the defaults easily we could have just two structs.
```
ConnectionOptions struct {}
APIRouteOptions struct {}
```

I'm thinking that there are a number of common API parameters that could be included in the `APIRoutes` eventually, like `async` `root` `force` etc. 

Then the defaults can be defined as 
```go
DefaultGetSkyKeyOptions = APIRouteOptions {
  ConnectionOptions: defaultConnectionOptions,
  APIRoute: "/skynet/skykey",
}
```